src/Nox.Solution/NoxCodeGenConventions.cs

```diff
@@ -60,5 +60,5 @@ public NoxCodeGenConventions(NoxSolution noxSolution)
 
     public string GetEntityDtoTypeFullName(string dtoName) => $"{DtoNameSpace}.{dtoName}";
 
-    public static string GetForeignKeyPropertyName(string foreignEntityName) => $"{foreignEntityName}Id";
+    public static string GetForeignKeyPropertyName(Entity entity, EntityRelationship relationship) => $"{entity.GetNavigationPropertyName(relationship)}Id";
```

Comment:
This is now computing the NavigationProperty name and not the ForeignKey name (Entity used as a key attribute)?